pwiz_tools/Skyline/TestUtil/ScreenshotManager.cs

```diff
@@ -0,0 +1,309 @@
+                    new Point(0, 0), shotFrame.Size);
+                graphCapture.Dispose();
+                return bmCapture;
+            }
+
+            [NotNull]
+            public override XmlNode Serialize(XmlDocument pDoc)
+            {
+                XmlNode node = pDoc.CreateElement(SHOT_ELEMENT);
+                XmlAttribute typeAttr = pDoc.CreateAttribute(SHOT_TYPE_ATTRIBUTE);
+                typeAttr.Value = SHOT_TYPE_VAL_ACTIVE_FORM;
+                node.Attributes.Append(typeAttr);
+                pDoc.DocumentElement.AppendChild(node);
+                return node;
+            }
+
+        }
+
+        private class CustomAreaShot : SkylineScreenshot
+        {
+            private Rectangle _shotFrame;
+            public CustomAreaShot(SkylineWindow pSkylineWindow, [NotNull] XmlNode pNode) : base(ShotType.ActiveWindow, pSkylineWindow)
+            {
+                if (pNode.FirstChild != null && pNode.FirstChild.LocalName == SHOT_FRAME_ELEMENT)
+                {
+                    XmlAttributeCollection rAtts = pNode.FirstChild.Attributes;
+                    _shotFrame = new Rectangle(Int16.Parse(rAtts[SHOT_FRAME_LEFT_ATTRIBUTE].Value),
+                        Int16.Parse(rAtts[SHOT_FRAME_TOP_ATTRIBUTE].Value),
+                        Int16.Parse(rAtts[SHOT_FRAME_RIGHT_ATTRIBUTE].Value),
+                        Int16.Parse(rAtts[SHOT_FRAME_BOTTOM_ATTRIBUTE].Value)
+                    );
+                }
+                else throw new InvalidDataException("Expected frame coordinates for this type of a screenshot, but it was not found.");
+            }
+            public CustomAreaShot(SkylineWindow pSkylineWindow) : base(ShotType.SkylineCustomArea, pSkylineWindow)
+            {
+                
+            }
+
+            [NotNull]
+            public override XmlNode Serialize(XmlDocument pDoc)
+            {
+                throw new NotImplementedException();
+            }
+
+            /**
+             * Display the snapshot area selector and return the resulting rectangle
+             * in display coordinates.
+             */
+            public Rectangle ShowAreaSelector()
+            {
+                throw new NotImplementedException();
+            }
+
+            public override void SetUp()
+            {
+            }
+
+            public override Bitmap Take(Form activeWindow)
+            {
+                throw new NotImplementedException();
+            }
+
+        }
+
+        [DllImport("user32.dll")]
+        public static extern bool SetProcessDPIAware();
+
+        private string filePath
+        {
+            get
+            {
+                var exeDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+                return Path.Combine(exeDir ?? "", _tutorialName + "_shots");
+            }
+        }
+
+        public ScreenshotManager([NotNull] string pTutorialName, [NotNull] SkylineWindow pSkylineWindow)
```

Comment:
We never put "p" at the front of variable names.

Reply:
Replaced this parameter with TestContext anyways.